src/components/Circles/Thread.tsx

```diff
@@ -1,52 +1,124 @@
-    return comments.flatMap((comment) => entriesForPost(comment));
-  }, [replies]);
+  const handleCreatePostClosed = useCallback((createdNewPost?: boolean) => {
+    setIsCreateCommentOpen(false);
+    setShouldScrollToEnd(!!createdNewPost);
+  }, []);
 
-  return (
-    <FlatList
-      testID="post-details-list"
-      data={entries}
-      renderItem={renderItem}
-      refreshing={isLoading || isRefetching}
-      onRefresh={postIn ? undefined : refetch}
-      initialNumToRender={35}
-      style={styles.container}
-      ListHeaderComponent={!error && post ? <ThreadPost post={post} /> : null}
-      ListEmptyComponent={
-        <View
-          style={styles.emptyPostCommentsContainer}
-          testID="empty-state-post-list-details"
-        >
-          {error ? <PostUnavailable /> : <EmptyComments />}
-        </View>
+  const comments = replies?.map(({ node }) => node).reverse() ?? [];
+  const entries = comments.flatMap((comment) => entriesForPost(comment));
+
+  const handleContentSizeChanged = useCallback(() => {
+    if (shouldScrollToEnd) {
+      setShouldScrollToEnd(false);
+      try {
+        listRef.current?.scrollToIndex({
+          index: entries.length - 1,
+          viewOffset: 100,
+        });
+      } catch {
+        // Scrolling failed but that's okay.
       }
-    />
+    }
+  }, [shouldScrollToEnd, entries.length]);
+
+  return (
+    <>
+      <FlatList
+        testID="post-details-list"
+        data={entries}
+        ref={listRef}
+        renderItem={renderItem({ onOpenThread })}
+        refreshControl={
+          <RefreshControl
+            onRefresh={refetch}
+            refreshing={isFetched && (isLoading || isRefetching)}
+            colors={[colors.primarySource]}
+            tintColor={colors.inversePrimary}
+          />
+        }
+        onContentSizeChange={handleContentSizeChanged}
+        initialNumToRender={35}
+        style={styles.container}
+        ListHeaderComponent={
+          !error && post ? <ThreadPost post={post as Post} /> : null
+        }
+        ListFooterComponent={
+          !isFetched && (isLoading || isRefetching) ? (
+            <ActivityIndicatorView
+              message={t('thread-screen-loading-comments', 'Loading comments')}
+            />
+          ) : null
+        }
+        ListEmptyComponent={
+          isLoading || isRefetching ? null : (
+            <View
+              style={styles.emptyPostCommentsContainer}
+              testID="empty-state-post-list-details"
+            >
+              {error ? <PostUnavailable /> : <EmptyComments />}
+            </View>
+          )
+        }
+      />
+
+      <TouchableOpacity
+        style={styles.addCommentBox}
+        onPress={() => setIsCreateCommentOpen(true)}
+      >
+        <Text style={styles.addCommentText}>
+          {t('thread-screen-add-comment', 'Add a comment')}
+        </Text>
+      </TouchableOpacity>
```

Comment:
Do we want to use Buttons and Text from react-native-paper in places like this?

I think we want to use Paper components as much as possible to make our own code cleaner and more themeable.

Reply:
Good point. I switched to use RNPaper's Text but left it as touchable opacity since it is kind of a unique button:
![image](https://github.com/lifeomic/react-native-sdk/assets/2295908/cdcc2bc7-d73b-4b2c-b70b-bcd3209556aa)